src/test/java/com/puppycrawl/tools/checkstyle/internal/XpathRegressionTest.java

```diff
@@ -212,53 +213,103 @@ public void validateIntegrationTestClassNames() throws Exception {
 
     @Test
     public void validateInputFiles() throws Exception {
-        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(inputDir)) {
+        try (DirectoryStream<Path> dirs = Files.newDirectoryStream(
+                inputDir,
+                path -> {
+                    return path.toFile().isDirectory();
+                }
+        )) {
             for (Path dir : dirs) {
-                // input directory must be named in lower case
-                assertWithMessage(dir + " is not a directory")
-                        .that(Files.isDirectory(dir))
-                        .isTrue();
-                final String dirName = dir.toFile().getName();
-                assertWithMessage("Invalid directory name: " + dirName)
-                        .that(ALLOWED_DIRECTORY_AND_CHECKS)
-                        .containsKey(dirName);
-
-                // input directory must be connected to an existing test
-                final String check = ALLOWED_DIRECTORY_AND_CHECKS.get(dirName);
-                final Path javaPath = javaDir.resolve("XpathRegression" + check + "Test.java");
-                assertWithMessage("Input directory '" + dir
-                            + "' is not connected to Java test case: " + javaPath)
-                        .that(Files.exists(javaPath))
-                        .isTrue();
-
-                // input files should be named correctly
-                validateInputDirectory(dir);
+                validateInputFiles(dir, dir.getFileName().toString());
```

Comment:
should be like:
`validateInputFiles(dir);`

Reply:
validateInputDir(dir)